src/Functions/FunctionBase64Conversion.cpp

```diff
@@ -0,0 +1,93 @@
+#include "config.h"
+#if USE_BASE64
+#    include <span>
+#    include <vector>
+#    include <Core/TypeId.h>
+
+namespace DB
+{
+
+std::vector<UInt8> preprocessBase64Url(const std::span<const UInt8> src)
+{
+    std::vector<UInt8> padded_src{};
```

Comment:
Let's omit `{}`.

Reply:
Alternatively, consider using `std::string` instead of `std::vector<UInt8>`. We could then call `operator+` or `.append("===")` in l. 49 - 64 which would be a bit more high-level.